course3/homework-vector/vector.cc

```diff
@@ -1,11 +1,45 @@
 #include <iostream>
 #include <cassert>
+#include <cmath>
 
 class Vector
 {
-    // TODO
+    int mX, mY;
+
+public:
+    Vector() {}
+    Vector(int x, int y) : mX(x), mY(y) {}
+
+    int GetX() const { return mX; }
+    int GetY() const { return mY; }
+
+    double GetPolarLength() const { return sqrt((mX * mX) + (mY * mY)); }
+    double GetPolarAngle() const { return atan2(mX, mY); }
+
+    void SetX(int x) { mX = x; }
+    void SetY(int y) { mY = y; }
+
+    bool operator==(const Vector& rhs) {
+        return ((rhs.GetX() == this->GetX()) && (rhs.GetY() == this->GetY()));
+    }
+
+    bool operator!=(const Vector& rhs) {
+        return !((rhs.GetX() == this->GetX()) && (rhs.GetY() == this->GetY()));
+    }
+
+    Vector operator+(const Vector& rhs) {
```

Comment:
This method can be const because the plus operator is not changing anything in the arguments of in the class. It's creating a new class with the summed X,Y